aten/src/ATen/native/cuda/DepthwiseConv3d.cu

```diff
@@ -0,0 +1,624 @@
+
+      for (int k_frame = 0; k_frame < kT; ++k_frame) {
+        const int out_frame_raw = out_frame_end - k_frame * dilationT;
+        const int out_frame = out_frame_raw / strideT;
+        for (int k_row = 0; k_row < kH; ++k_row) {
+          const int out_row_raw = out_row_end - k_row * dilationH;
+          const int out_row = out_row_raw / strideH;
+          for (int k_col = 0; k_col < kW; ++k_col) {
+            const scalar_t op1 = __ldg(kernel_ptr++);
+            const int out_col_raw = out_col_end - k_col * dilationW;
+            const int out_col = out_col_raw / strideW;
+
+            const int out_offs = (out_frame * oH + out_row) * oW + out_col;
+
+            scalar_t op2 = (scalar_t)0;
+            if (out_col >= 0 && out_row >= 0 && out_frame >= 0 &&
+                out_col < oW && out_row < oH && out_frame < oT) {
+              op2 = __ldg(gout_ptr + out_offs);
+            }
+            if (out_frame * strideT == out_frame_raw &&
+                out_row * strideH == out_row_raw &&
+                out_col * strideW == out_col_raw) {
+              sum += op1 * op2;
+            }
+          }
+        }
+      }
+    }
+
+    grad_input[batch][in_channel][in_frame][in_row][in_col] = sum;
+  }
+}
+
+template <typename scalar_t, typename accscalar_t,
+    int kKnownStrideH, int kKnownStrideW>
+__global__ void
+conv_depthwise3d_cuda_backward_weight_kernel(
+    const PackedTensorAccessor32<scalar_t, 5> grad_output,
+    const PackedTensorAccessor32<scalar_t, 5> input,
+    PackedTensorAccessor32<scalar_t, 5> grad_kernel,
+    int strideT, int strideH_, int strideW_,
+    int paddingT, int paddingH, int paddingW,
+    int dilationT, int dilationH, int dilationW) {
+  const int kC = grad_kernel.size(0);
+  const int kT = grad_kernel.size(2);
+  const int kH = grad_kernel.size(3);
+  const int kW = grad_kernel.size(4);
+
+  const int strideH = kKnownStrideH > 0 ? kKnownStrideH : strideH_;
+  const int strideW = kKnownStrideW > 0 ? kKnownStrideW : strideW_;
+
+  const int k_col = blockIdx.x % kW;
+  const int k_row = (blockIdx.x / kW) % kH;
+  const int k_frame = (blockIdx.x / kW / kH) % kT;
+  const int k_channel = blockIdx.x / kW / kH / kT;
+  scalar_t *result = &grad_kernel[k_channel][0][k_frame][k_row][k_col];
+
+  const int oT = grad_output.size(2);
+  const int oH = grad_output.size(3);
+  const int oW = grad_output.size(4);
+  const int iT = input.size(2);
+  const int iH = input.size(3);
+  const int iW = input.size(4);
+  const int channel_multiplier = grad_output.size(1) / input.size(1);
+  const int in_channel = k_channel / channel_multiplier;
+
+  extern __shared__ int sdata_raw[];
+  scalar_t* sdata = reinterpret_cast<scalar_t*>(sdata_raw);
+
+  if (k_channel >= kC) {
+    return;
+  }
+
+  const int laneid = threadIdx.x % C10_WARP_SIZE;
+  const int warpid = threadIdx.x / C10_WARP_SIZE;
+  const int nwarps = blockDim.x / C10_WARP_SIZE;
+
+  scalar_t grad = 0;
```

Comment:
should be accscalar_t. Also, it would be better to convert op1 and op2 to accscalar_t before multiplying them to avoid precision loss (similar in other kernels, you are doing multiplication in scalar_t, and that's prone to over/underflow and loss of precision).